### DISCUSSION TITLE
Application vs Compose

### DISCUSSION THREAD

**Discussion by kerimovok** [Dokploy/dokploy#791] [Q&A] [answered]
Hi,

I have been using 'Compose' type for my Docker Compose project for a long time, and I didn't understand why 'Application' type was needed.

Now, when I tried it, it seems to have **A LOT** more functionality (basic security, traefik config, swarm settings, different volume mount types) than 'Compose' type, and it works exactly the same way.

So my question is what is the real difference between them for Docker Compose projects specifically?

Thank you, for this great project!

**Comment by Siumauricio** [comment, marked as answer]
Hi,

The comparison is simple, 

**Docker compose**: you can have multiple services/containers inside it, here we use tipically `docker-compose.yml`
**Application**: you can only have 1 container and it is isolated in docker swarm, here we use tipically `Dockerfile`

Why application has more features? because initially all the features were designed for applications, in docker compose it is possible, however I don't have time to implement them but it is possible, almost everything that can be done in applications can be done in docker compose, but in docker compose you need to do it manually.

In my case I like applications better because as they are based on docker swarm I can have zero downtime, rollbacks and many things that docker compose does not.

**Reply by kerimovok** [reply]
Thank you very much for your clear response!

**Reply by kerimovok** [reply]
Also, I noticed that the 'application' automatically added to the docker network, which is super-super useful

I am just writing it, for informational purposes, just to note

**Reply by alirizatuncer** [reply]
Thanks for your explanation 👏🏼
